src/rules/dto-decorators.ts

```diff
@@ -0,0 +1,221 @@
+import { Rule, SourceCode } from "eslint";
+import {
+  ClassDeclaration,
+  Directive,
+  ModuleDeclaration,
+  Program,
+  Statement,
+  Node,
+  ExportDefaultDeclaration,
+} from "estree";
+import { findImportsByPackageName } from "../utils/imports-finder";
+
+interface ProficoNodeBodyElement {
+  decorators?: ProficoDecorator[];
+  type: "PropertyDefinition";
+}
+
+interface ProficoDecorator {
+  expression: {
+    callee: {
+      name: string;
+    };
+  };
+  type: "Decorator";
+  range: number[];
+}
+
+const dtoDecorators: Rule.RuleModule = {
+  meta: {
+    fixable: "code",
+    type: "problem",
+    messages: {
+      improperlyOrderedDecorators: "Decorators are not properly ordered.",
+    },
+    docs: {
+      url: "https://github.com/profico/eslint-plugin-profico#dto-decorators",
+    },
+  },
+  create: context => {
+    return {
+      Program({ body }: Program) {
+        const optionalDecorator = new Set(["IsOptional"]);
```

Comment:
It would be good to extract the "IsOptional" magic string into a constant since it's used multiple times and it's easy to make a lowercase/uppercase mistake.

Reply:
I'll create a constant.